### DISCUSSION TITLE
Is there any way to run multiple jmx configurations in a single JMeterRunTask?

### DISCUSSION THREAD

**Discussion by hanchenjhc** [qualersoft/jmeter-gradle-plugin#26] [Q&A] [answered]
It would be nice to run all jmx configurations in a directory as part of a CI/CD pipeline. Does the plugin support that? As far as I can see, jmxFile takes a single string. I tried jmxFile.set("*") that does't seem to be supported at the moment.

**Comment by mathze** [comment, marked as answer]
Hi @hanchenjhc,
nope, this isn't possible. Mainly because of the implicit coupling to report task. But I will think about it.
But maybe the next release will help you a bit, it exposes some of the tasks configuration properties to the cli so you could run same task multiple times and just alter the jmxFile.

Thank you for your input

**Reply by hanchenjhc** [reply]
Thanks for the reply!

**Reply by hanchenjhc** [reply]
Would it be possible for jmxFile to be a list and just do everything you do now to all the test files instead? Possibly generate a report per test file?

**Reply by mathze** [reply]
Had a closer look on this. Unfortunately, it isn't that easy. Cause the base task used, is a JavaExec task which is not designed to run in loop. Beside the issues of loose coupling between run and report task, it would require my to rewrite the whole base how the run- and report tasks work. So for now, I will not support this feature.

But maybe these ideas may help you:
1. Create a 'root' jmx-file that bundles your other jmx-files with Include-Controller
2. Use shell 
2.1) create a 'run'-task that setups anything required to run your tests but leave jmx-File empty
2.2) use shell to get all jmx-File and pipe them to `./gradlew 'run' --test $foundJmxFile`
3. You could also dynamically create run-tasks within the gradle-script and create a simple task that depends on all the created run tasks

First approach is quite inflexible because you always have to update the 'root' jmx-file if you add new other jmx-files.
Second approach seems to me the most elegant but it came with the downside that you may have a lock on OS because of used shell commands.
Third approach is platform independent but will clutter your build-script with many run-(/report-)tasks. I personally wouldn't do it unless there is no other way ;)

**Reply by hanchenjhc** [reply]
I see. Thanks for the suggestions!
